File: dbflow/src/main/java/com/raizlabs/android/dbflow/sql/migration/BaseMigration.java
package com.raizlabs.android.dbflow.sql.migration;

import androidx.annotation.NonNull;

import com.raizlabs.android.dbflow.structure.database.DatabaseWrapper;

/**
 * Description: Provides the base implementation of {@link com.raizlabs.android.dbflow.sql.migration.Migration} with
 * only {@link Migration#migrate(DatabaseWrapper)} needing to be implemented.
 */
public abstract class BaseMigration implements Migration {


    @Override
    public void onPreMigrate() {

    }

    @Override
    public abstract void migrate(@NonNull DatabaseWrapper database);

    @Override
    public void onPostMigrate() {

    }
}

File: dbflow/src/main/java/com/raizlabs/android/dbflow/sql/migration/BaseMigration.kt
package com.raizlabs.android.dbflow.sql.migration

import com.raizlabs.android.dbflow.structure.database.DatabaseWrapper

/**
 * Description: Provides the base implementation of [com.raizlabs.android.dbflow.sql.migration.Migration] with
 * only [Migration.migrate] needing to be implemented.
 */
abstract class BaseMigration : Migration {


    override fun onPreMigrate() {

    }

    abstract override fun migrate(database: DatabaseWrapper)

    override fun onPostMigrate() {

    }
}
